firmware/logging.h

```diff
@@ -26,20 +26,25 @@
 #include "chprintf.h"
 
 
-enum logging_levels
-{
+enum logging_levels {
     logging_level_none,
     logging_level_info,
     logging_level_trace
 };
 
+#define log_info(msg, ...) if (get_logging_level() >= logging_level_info) {chprintf((BaseSequentialStream *)&SD1, "%i ", ST2MS(chVTGetSystemTimeX())); chprintf((BaseSequentialStream *)&SD1, msg, ##__VA_ARGS__);}
+#define log_trace(msg, ...) if (get_logging_level() >= logging_level_trace) {chprintf((BaseSequentialStream *)&SD1, "%i ", ST2MS(chVTGetSystemTimeX())); chprintf((BaseSequentialStream *)&SD1, msg, ##__VA_ARGS__);}
 
-#define log_info(msg, ...) if (get_logging_level() >= logging_level_info) {chprintf((BaseSequentialStream *)&SD1, msg, ##__VA_ARGS__);}
-#define log_trace(msg, ...) if (get_logging_level() >= logging_level_trace) {chprintf((BaseSequentialStream *)&SD1, msg, ##__VA_ARGS__);}
-//#define debug_write
+/* Brief functions */
+#define log_info_b(msg, ...) if (get_logging_level() >= logging_level_info) {chprintf((BaseSequentialStream *)&SD1, msg, ##__VA_ARGS__);}
```

Comment:
I had a problem passing the va_args via functions, that's why I did the #defines. Maybe we can chat on how to do this via a function.  I implemented the functions using   , ...) as the va_arg list and the compiler didn't like it.